#!/usr/bin/env python
PACKAGE = "mav_linear_mpc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("q_x",    double_t,    0, "Penality on x axis pos", 50.0 ,  0.001, 200.0)
gen.add("q_y",    double_t,    0, "Penality on y axis pos", 50.0 ,  0.001, 200.0)
gen.add("q_z",    double_t,    0, "Penality on z axis pos", 80.0 ,  0.001, 200.0)

gen.add("q_vx",    double_t,    0, "Penality on x axis vel", 20.0 ,  0.001, 200.0)
gen.add("q_vy",    double_t,    0, "Penality on y axis vel", 20.0 ,  0.001, 200.0)
gen.add("q_vz",    double_t,    0, "Penality on z axis vel", 35.0 ,  0.001, 200.0)

#gen.add("q_roll",     double_t,    0, "Penality on roll state", 20.0 ,  0.001, 200.0)
#gen.add("q_pitch",    double_t,    0, "Penality on pitch state", 20.0 ,  0.001, 200.0)

gen.add("r_thrust_x",   double_t,    0, "Penality on thrust x cmd", 1.0 ,  0.1, 200)
gen.add("r_thrust_y",   double_t,    0, "Penality on thrust y cmd", 1.0 , 0.1, 200)
gen.add("r_thrust_z",   double_t,    0, "Penality on thrust z cmd", 1.0 , 0.1, 200)

gen.add("r_dthrust_x",   double_t,    0, "Penality on delta thrust x cmd", 0.002 ,  0.001, 0.1)
gen.add("r_dthrust_y",   double_t,    0, "Penality on delta thrust y cmd", 0.002 ,  0.001, 0.1)
gen.add("r_dthrust_z",   double_t,    0, "Penality on delta thrust z cmd", 0.002,  0.001, 0.1)

gen.add("thrust_x_max",     double_t,   0, "max thrust x cmd [m/s2]", 1.0, 0.5, 3.0)
gen.add("thrust_y_max",     double_t,   0, "max thrust y cmd [m/s2]", 1.0, 0.5, 3.0)
gen.add("thrust_z_max",     double_t,   0, "max thrust z cmd [m/s2]", 13.0, 11.0, 14.0)
gen.add("thrust_z_min",     double_t,   0, "max thrust z cmd [m/s2]", 7.0 ,  6.0, 9.0)

gen.add("rot_acc_x_max",     double_t,   0, "max rot acc x cmd [m/s2]", 1.0, 0.5, 2.0)
gen.add("rot_acc_y_max",     double_t,   0, "max rot acc y cmd [m/s2]", 1.0, 0.5, 2.0)
gen.add("rot_acc_z_max",     double_t,   0, "max rot acc z cmd [m/s2]", 1.0, 0.5, 2.0)

gen.add("KR_x",   double_t,    0, "KR x", 1 ,  0.01, 10.0)
gen.add("KR_y",   double_t,    0, "KR y", 1 ,  0.01, 10.0)
gen.add("KR_z",   double_t,    0, "KR z", 1 ,  0.01, 10.0)

gen.add("K_omega_x",   double_t,    0, "K omega x", 1 ,  0.01, 10.0)
gen.add("K_omega_y",   double_t,    0, "K omega y", 1 ,  0.01, 10.0)
gen.add("K_omega_z",   double_t,    0, "K omega z", 1 ,  0.01, 10.0)

gen.add("K_yaw",   double_t,    0, "yaw gain", 0.5 ,  0.01, 2.0)
gen.add("Ki_xy",   double_t,    0, "integrator gain for xy", 0.2 ,  0.01, 3.0)
gen.add("Ki_altitude",   double_t,    0, "integrator gain for altitude", 0.2 ,  0.01, 3.0)

gen.add("enable_offset_free",   bool_t,   0, "XY offset free MPC",  True)
gen.add("enable_integrator",   bool_t,   0, "integral action",  False)

exit(gen.generate(PACKAGE, "Config", "LinearMPC"))
